extensions/vertx-core/runtime/src/main/java/io/quarkus/vertx/core/runtime/VertxCoreRecorder.java

```diff
@@ -259,9 +260,13 @@ private static VertxOptions convertToVertxOptions(VertxConfiguration conf, Vertx
                 if (!tmp.mkdirs()) {
                     LOGGER.warnf("Unable to create Vert.x cache directory : %s", tmp.getAbsolutePath());
                 }
-                if (!(tmp.setReadable(true, false) && tmp.setWritable(true, false))) {
-                    LOGGER.warnf("Unable to make the Vert.x cache directory (%s) world readable and writable",
-                            tmp.getAbsolutePath());
+                String os = System.getProperty("os.name").toLowerCase(Locale.ENGLISH);
+                if (!os.contains("windows")) {
+                    // Do not execute the following on windows.
+                    if (!(tmp.setReadable(true, false) && tmp.setWritable(true, false))) {
+                        LOGGER.warnf("Unable to make the Vert.x cache directory (%s) world readable and writable",
```

Comment:
Unrelated question but why are we trying to make it world writable? Shouldn't it be only writable by the current user? Am I missing something?

Reply:
There is a structure. The first level is world writable as multiple Quarkus app started by multiple users will write into it. The nested level is limited to the user.